picocoap/src/coap.c

```diff
@@ -247,13 +247,15 @@ coap_error coap_init_pdu(coap_pdu *pdu)
 	if (pdu->max < 4)
 		return CE_INSUFFICIENT_BUFFER;
 
+	pdu->len = 0;
+	memset(pdu->buf, 0, 4);
```

Comment:
It's the length of data that will be in the buffer after the basic header is created. Would it be valuable to create a COAP_HRD_NO_TOKEN_LEN constant? Or would a comment there be good enough to clarify?